mobilecoind/src/payments.rs

```diff
@@ -179,17 +216,17 @@ impl<T: BlockchainConnection + UserTxConnection + 'static, FPR: FogPubkeyResolve
     ) -> Result<(u64, u32), Error> {
         // Figure out the block_version and fee, taking into account if opt_fee is
         // nonzero
-        let candidate_block_version = self.ledger_db.get_latest_block()?.version;
+        let block_version = max(
+            self.ledger_db.get_latest_block()?.version,
+            last_block_info.network_block_version,
+        );
+
         Ok(if opt_fee != 0 {
-            (opt_fee, candidate_block_version)
+            (opt_fee, block_version)
         } else {
             let fee = last_block_info
                 .minimum_fee_or_none(&token_id)
-                .unwrap_or(FALLBACK_FEE);
-            let block_version = max(
-                candidate_block_version,
-                last_block_info.network_block_version,
-            );
+                .ok_or_else(|| Error::TxBuild("Token cannot be used to pay fees".into()))?;
```

Comment:
this is what we do now instead of the `FALLBACK_FEE`

it's possible that we should give this error if the token id is not in the last_block_info, even if the user provided a fee value

Reply:
yes, in the scenario that there is network congestion, users have to pay higher fees to get their transactions through

Reply:
Yes, it will happily take a larger fee. Which is why this FALLBACK_FEE was problematic - users might end up unexpectedly paying higher fees than necessary